src/Lean/Elab/Term.lean

```diff
@@ -1891,7 +1891,7 @@ def addAutoBoundImplicits' (xs : Array Expr) (type : Expr) (k : Array Expr → E
     forallBoundedTelescope (← mkForallFVars xs type) xs.size fun xs type => k xs type
 
 def mkAuxName (suffix : Name) : TermElabM Name := do
-  match (← read).declName? with
+  match (← read).declName? <|> (← getEnv).asyncPrefix? with
```

Comment:
Is there some invariant here at play? declName always being inside the asyncPrefix maybe?

Reply:
Yes, that invariant should be correct. We really should clean up this aux name function duplication mess but for now I just want to get in the smallest possible fixes.

Reply:
Right, maybe leave a comment here (and maybe later the function that changes the elaborate declName can check this invariant?)